sycl/include/CL/sycl/detail/usm_dispatch.hpp

```diff
@@ -0,0 +1,73 @@
+//==-------------- usm_dispatch.hpp - SYCL USM Dispatch --------*- C++ -*---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// ===--------------------------------------------------------------------=== //
+#pragma once
+
+#include <CL/sycl/detail/clusm.hpp>
+
+#include <memory>
+
+namespace cl {
+namespace sycl {
+namespace detail {
+namespace usm {
+
+class USMDispatcher {
+public:
+  USMDispatcher(cl_platform_id Platform);
+  
+  void *hostMemAlloc(pi_context Context, cl_mem_properties_intel *Properties,
+                     size_t Size, pi_uint32 Alignment, pi_result *ErrcodeRet);
+  void *deviceMemAlloc(pi_context Context, pi_device Device,
```

Comment:
Are you going to move these functions to pi.hpp/pi.cpp ?

Reply:
No, I think it's better to keep them here as we need some state (the CLUSM object).  In some sense the Dispatcher is separate piece of PI.  It's easy to wire new backends up here.